app/models/concerns/framework_assessmentable.rb

```diff
@@ -0,0 +1,159 @@
+module FrameworkAssessmentable
+  extend ActiveSupport::Concern
+  include StateMachineable
+
+  ASSESSMENT_NOT_STARTED = 'not_started'.freeze
+  ASSESSMENT_IN_PROGRESS = 'in_progress'.freeze
+  ASSESSMENT_COMPLETED = 'completed'.freeze
+  ASSESSMENT_CONFIRMED = 'confirmed'.freeze
+
+  included do |assessment_klass|
+    # "not_started" cannot be used as the name of the enum due to warnings in the model
+    # that it starts with a "not_".
+    enum statuses: {
+      unstarted: ASSESSMENT_NOT_STARTED,
+      in_progress: ASSESSMENT_IN_PROGRESS,
+      completed: ASSESSMENT_COMPLETED,
+      confirmed: ASSESSMENT_CONFIRMED,
+    }
+
+    validates :status, presence: true, inclusion: { in: statuses }
+    validates :profile, uniqueness: true
+    validates :confirmed_at, presence: { if: :confirmed? }
+
+    has_many :framework_responses, as: :assessmentable, dependent: :destroy
+    has_many :generic_events, as: :eventable, dependent: :destroy # NB: polymorphic association
+
+    belongs_to :framework
+    has_many :framework_questions, through: :framework
+    has_many :framework_flags, through: :framework_responses
+    belongs_to :profile
+    belongs_to :move, optional: true
+    belongs_to :prefill_source, class_name: assessment_klass.to_s, optional: true # rubocop:disable Rails/ReflectionClassName
```

Comment:
Not sure how to get around this without disabling, see https://github.com/rubocop-hq/rubocop-rails/issues/106

Reply:
move out to specific models